optimum/exporters/onnx/__main__.py

```diff
@@ -115,41 +122,65 @@ def main():
             f"At least  {onnx_config.DEFAULT_ONNX_OPSET} is required."
         )
 
-    onnx_inputs, onnx_outputs = export(
-        model,
-        onnx_config,
-        args.opset,
-        args.output,
-    )
+    if args.atol is None:
+        args.atol = onnx_config.ATOL_FOR_VALIDATION
+        if isinstance(args.atol, dict):
+            args.atol = args.atol[task.replace("-with-past", "")]
+
+    if model.config.is_encoder_decoder and args.for_ort:
+        encoder_model = model.get_encoder()
+        encoder_onnx_config = onnx_config.get_encoder_onnx_config(encoder_model.config)
+
+        decoder_model = model.get_decoder()
+        decoder_onnx_config = onnx_config.get_decoder_onnx_config(
+            decoder_model.config, task.replace("-with-past", ""), use_past=False
+        )
+
+        models_for_export = [
+            (encoder_model, encoder_onnx_config, "encoder_model.onnx"),
+            (model, decoder_onnx_config, "decoder_model.onnx"),
+        ]
+
+        if "-with-past" in task:
+            decoder_onnx_config_with_past = onnx_config.get_decoder_onnx_config(
+                decoder_model.config, task.replace("-with-past", ""), use_past=True
+            )
+            models_for_export.append((model, decoder_onnx_config_with_past, "decoder_with_past_model.onnx"))
+    else:
+        models_for_export = [(model, onnx_config, "model.onnx")]
```

Comment:
Would it be possible to add this logic inside a function, and to use this function in `export`, or in both `export_pytorch` and `export_tensorflow`?

Reply:
Could be possible to move to `export` itself. But along with this the `validate_model_outputs` function would also need to be updated support multiple model validation.

Reply:
You mean moving the validation inside the export function?

Reply:
Not exactly. If we the above logic for ORT inside the `export` function we would return 3 `onnx_outputs`. The same logic to create 3 models should also be part of the `validate_model_outputs`. Hence both the functions would change.

Reply:
Created separate functions to export and validate  encoder-decoder models for conditional generation. Found this to be more cleaner than adding new args to the existing export function.